src/processing/spot.rs

```diff
@@ -217,53 +218,83 @@ pub async fn process_long_tail_asset(
     pair: String,
     sources: Vec<String>,
 ) -> Result<u64, MonitoringError> {
+    let config = get_config(None).await;
+    let network_env = &config.network_str();
+    let decimals = *config.decimals(DataType::Spot).get(&pair).unwrap();
+
+    let mut timestamps = Vec::with_capacity(sources.len());
+    let mut deviations = Vec::with_capacity(sources.len());
+
+    for source in sources.iter() {
+        log::info!("Processing data for pair: {} and source: {}", pair, source);
+        let (timestamp, price_deviation) =
+            get_price_deviation_for_source_from_chain(pool.clone(), &pair, source, decimals)
+                .await?;
+        timestamps.push(timestamp);
+        deviations.push(price_deviation);
+    }
+
+    let threshold = get_long_tail_threshold(&pair, sources.len()).unwrap();
```

Comment:
no unwrap

Reply:
in that case just add a comment on top explaining why it's fine

Reply:
done